src/fges_helper.jl

```diff
@@ -0,0 +1,242 @@
+        return true
+    end
+
+    queue = [x]
+    visited[x] = true
+    while !isempty(queue)
+        currentNode = popfirst!(queue) # get new element from queue
+        for vᵢ in descendent(g,currentNode)
+            vᵢ == y && return false
+            if !visited[vᵢ]
+                push!(queue, vᵢ) # push onto queue
+                visited[vᵢ] = true
+            end
+        end
+    end
+    return true
+end
+
+
+
+####################################################################
+# Modify Edges
+####################################################################
+
+"""
+    orientedge!(g, x, y)
+Update the edge `x`-`y` to `x`→`y` in the graph `g`. 
+"""
+orientedge!(g, edge) = rem_edge!(g, reverse(edge))
+orientedge!(g, x, y) = rem_edge!(g, y, x)
+
+
+
+####################################################################
+# Neighborhood functions 
+####################################################################
+
+#Should these be non-allocating iterators? 
+neighborsGeneral(isFunction::Function, g, x) = Iterators.filter(vᵢ -> isFunction(g,vᵢ,x), vertices(g))
+
+#Note: inneighbors and outneighbors are already defined in Graphs.jl
+
+#All vertices connected to x by an undirected edge
+neighbors_undirect(g, x) = neighborsGeneral(isundirected, g, x)
+
+#these are just aliases for the functions above
+parents(g, x) = inneighbors(g, x)
+children(g, x) = outneighbors(g, x)
+
+descendent(g, x) = neighborsGeneral(isdescendent, g, x)
+
+
+####################################################################
+# Counting Functions
+####################################################################
+
+#Very similar to the neighborsGeneral function except we only keep a count of the valid nodes
+function countGeneral(isFunction::Function, g, x)
+    counter=0
+    for vᵢ in vertices(g)
+        if isFunction(g,vᵢ,x)
+            counter += 1
+        end
+    end
+    return counter
+end
+
+#All vertices connected to x
+countNeighbors(g, x) = countGeneral(isadjacent, g, x)
+
+#All vertices pointing to x
+countNeighbors_in(g, x) = countGeneral(isparent, g, x)
+
+#All vertices pointing away from x
+countNeighbors_out(g, x) = countGeneral(ischild, g, x)
+
+#All vertices connected to x by an undirected edge
+countNeighbors_undirect(g, x) = countGeneral(isundirected, g, x)
+
+#these are just aliases for the functions above
+countParents(g, x) = countNeighbors_in(g, x)
+countChildren(g, x) = countNeighbors_out(g, x)
+
+####################################################################
+# all* functions
+####################################################################
+
+#Generate an iterator equivalent to combinations(x,2) but produces a vector of tuples
+allpairs(v) = Iterators.filter(i -> isless(i...), Iterators.product(v,v))
+
+#Get every undirected edge in the entire graph g
+allundirected(g) = [Edge(nodePair) for nodePair in allpairs(vertices(g)) if isundirected(g,nodePair)]
```

Comment:
Make it a generator?

Reply:
Do you mean something like this?
```suggestion
allundirected(g) = (Edge(nodePair) for nodePair in allpairs(vertices(g)) if isundirected(g,nodePair))
```

Reply:
Yes.